### PR TITLE
Added build instructions for Ubuntu 16.04 and Docker

### PR DESCRIPTION
**Pull request by muisit**
This PR adds a doc directory containing build instructions for a basic Ubuntu 16.04 LTS platform.

It also contains a Dockerfile that sets up a build environment for the wallet based on the Ubuntu 16.04 instructions, containing an entry point script that allows the user to execute various build commands at the installed toolset (npm, ng, ionic). Using this docker, users can easily build Android files based on the wallet configurations.
I did not test this for XCode compatibility.

Work gratefully sponsored by SURFnet, Netherlands, https://surf.nl

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+\n\
+cd /data/wallet\n\
+case \$1 in\n\
+wait)\n\
+    sleep \$2\n\
+    ;;\n\
+init)\n\
+    npm install\n\
+\n\
+    cd node_modules/@angular-devkit/build-angular/src/angular-cli-files/models/webpack-configs\n\
```
Comment (rdemera) [inline Dockerfile:70]: this issue is solved in https://github.com/alastria/alastria-wallet/pull/78

File: doc/install_ubuntu1604.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+## Wallet Installation
+```
+cd $HOME
+git clone https://github.com/alastria/alastria-wallet.git
+
+cd alastria-wallet
+npm install
+ 
+# see: https://github.com/auth0/node-jsonwebtoken/issues/471#issuecomment-398798497
```
Comment (rdemera) [inline doc/install_ubuntu1604.md:69]: This issue is solved in https://github.com/alastria/alastria-wallet/pull/78

**Comment by VictorNS69** [comment]
Hi @muisit we would like to know if the issues you ask in this PR are already fixed.

If you consider that there are still some issues to be corrected, please let us know.

If there is no more activity in this PR we will close it soon.

**Comment by muisit** [comment]
This PR is about extending the Alastria code base with documentation about installing and building the project on Ubuntu 16, including a Dockerfile build file to create a containerised build environment. There are no 'issues to be fixed'. The only thing this PR fixes is 'missing documentation for building using Ubuntu 16'. 

I don't think #78 fixes anything that coincides with this PR. The only minor element that coincides with #78 is the fact that 'npm install' is being run. There is no other similarity at all.  I don't know why 'rdemera' tagged this PR, but probably because he saw that amongst the changes an 'npm install' was found. Whatever #78 fixes, it does not overlap with this PR.

I no longer have the opportunity to work on Alastria at the moment, so I cannot assist in updating or adjusting this PR any further. If you do not think it is worthwhile to add to the code base, please remove it.

**Pull request merged**